### PR TITLE
[FIVE 214] (Frontend) Obtener y Listar Productos de AWS

### PR DESCRIPTION
**Pull request by gonzalorom**
### Descripción
Como cliente, deseo poder ver los artefactos disponibles que puedo contratar a traves de Amazon AWS para la infraestructura de mi proyecto.
### Criterios de aceptacion:
- Al seleccionar como Tipo de artefacto a Artefacto AWS, mostrar una pagina de seleccion de artefacto.
- Listar todos los artefactos disponibles.
- Agregar filtros para reducir la cantidad de artefactos visibles en la lista (Campos nombre, tipo)
- Implementar paginado de 20 resultados por pagina.
### Cambios realizados:
- Se creo UI (combo box) en `AwsForm` para la seleccionar nombre del artefacto AWS.
- Se creo `AwsArtifactService` en capa client para gestionar llamadas a la api.
- Se creo `AwsArtifact` como interfaz para el manejo de datos desde la api.

### CODE REVIEW

**Comment by matiasbeckerle** [comment]
Diff seems to be including changes from the other Gonza as well.

**Comment by gonzalorom** [comment]
Yes, and I had only created this branch from release. The real changes are in that commit #8bfd69c.

File: FiveRockingFingers/FRF.Web/ClientApp/src/components/NewArtifactDialogComponents/AwsForm.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                {...params}
+                label="Nombre del artefacto AWS"
+                variant="outlined"
+                InputProps={{
+                  ...params.InputProps,
+                  endAdornment: (
+                    <React.Fragment>
+                      {loading ? (
+                        <CircularProgress color="inherit" size={20} />
+                      ) : null}
```
Comment (gperelstein) [inline FiveRockingFingers/FRF.Web/ClientApp/src/components/NewArtifactDialogComponents/AwsForm.tsx:136]: I like the idea of incorporating a circular loader!

File: FiveRockingFingers/FRF.Web/ClientApp/src/components/NewArtifactDialog.tsx [hunk 2]
```diff
@@ -1,164 +1,106 @@
-﻿import { Button, Dialog, DialogActions, DialogContent, DialogTitle, FormControl, FormHelperText, InputLabel, MenuItem, Select, TextField } from '@material-ui/core';
-import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
```
Comment (gperelstein) [inline FiveRockingFingers/FRF.Web/ClientApp/src/components/NewArtifactDialog.tsx:2]: I don't know why it happens but in this PR appeared some changes that have been done in  other PR

**Review by gperelstein — APPROVED** [review]
Everything is excellent for me!

File: FiveRockingFingers/FRF.Web/ClientApp/src/components/NewArtifactDialog.tsx (hunk 2 above)
Comment (gonzalorom) [reply FiveRockingFingers/FRF.Web/ClientApp/src/components/NewArtifactDialog.tsx]: Yes. there was a conflict between the source PR and the realese_01. Now that is gone.

File: FiveRockingFingers/FRF.Web/ClientApp/src/components/NewArtifactDialogComponents/AwsForm.tsx (hunk 1 above)
Comment (gonzalorom) [reply FiveRockingFingers/FRF.Web/ClientApp/src/components/NewArtifactDialogComponents/AwsForm.tsx]: Yes, I try to use it as much as I can. I feel like gives some kind of relief to the user who is using the application and waiting for the answer. Like the music in the elevators.

**Pull request merged**